### PR TITLE
Run tests on iOS using Mac Catalyst and expand Apple platforms

### PR DESCRIPTION
**Pull request by complexspaces**
This PR accomplishes two things:
- Compiling and running all of the tests on an iOS target, by utilizing Catalyst targets as pointed out in [today's 1.82.0 release notes](https://blog.rust-lang.org/2024/10/17/Rust-1.82.0.html). 
    - This is not quite as complete as if we ran these tests in a full simulator, but the environment isn't going to be that different in our use case where we are solely calling `Security.framework` functions. Even in the simulator, a lot of frameworks and kernel functions come "from the macOS host" anyway.
- Expanding the Apple platforms that `rustls-platform-verifier` can be compiled for to all available ones today.
    - AFAICT, all of the `Security.framework` functions we call are available everywhere and have no special OS constraints. All of them make sense to support as well:
    - visionOS is very similar to iOS and has the same networking API availability, etc.
    - watchOS has some network API constraints which restrict what you could perform TLS over, but you could still bring your own transport encryption stack as [per the requirement documentation](https://developer.apple.com/documentation/technotes/tn3135-low-level-networking-on-watchos).

Closes https://github.com/rustls/rustls-platform-verifier/issues/4 (unless anyone thinks we need the simulator explicitly)

### CODE REVIEW

**Comment by complexspaces** [comment]
This PR is blocked on GitHub updating their runner image to move from Rust 1.81 to 1.82 for a "stable" tier 2 Catalyst target.

File: rustls-platform-verifier/Cargo.toml
Comment (djc) [inline rustls-platform-verifier/Cargo.toml:39]: Nit: feels weird to have a first commit that adds `target_os = "ios"` only to then have a commit that replace the Apple `target_os` attributes with `target_vendor = "apple"`, maybe we should do it the other way around and replace with `target_vendor` first, then add specific `ios` stuff where needed?

Comment (complexspaces) [reply rustls-platform-verifier/Cargo.toml:39]: I originally had this all in one commit but decided to split it out 😅. I can flip the order since you think its better that way.

Comment (djc) [reply rustls-platform-verifier/Cargo.toml:39]: Yeah, I usually like to minimize within-PR churn between commits.

Comment (complexspaces) [reply rustls-platform-verifier/Cargo.toml:39]: I've force pushed a re-arrangement of the commits so the Catalyst testing builds ontop of the `target_vendor = "apple"` base instead. No rush to re-review though since it will probably be next week when GitHub bumps their Rust versions.

**Review by cpu — APPROVED** [review]
LGTM once the runners update and CI passes. Very nice to get this coverage :rocket:

**Review by djc — APPROVED** [review]

**Comment by complexspaces** [comment]
The update finally rolled out so this one was ready 👍

**Pull request merged**
